extension/js/common/api/api.ts

```diff
@@ -806,4 +806,8 @@ export class Api {
     apiFcCall: (path: string, vals: Dict<any>, fmt: ReqFmt = 'JSON') => Api.internal.apiCall(Api.fc.url('api'), path, vals, fmt, undefined, { 'api-version': '3' }),
   };
 
+  public static retreiveBlogPosts = async () => {
```

Comment:
Looks good, just add the types as discussed above, follow the style of other API methods.